### PR TITLE
113 front improvement

### LINKED ISSUE #113: Front improvement
**Short description:**
This issue implement all multiple front things add to create a beautiful app.

**Describe the solution you'd like**
Implement so many things :sweat_smile:

### LINKED ISSUE #129: An AREA can have several Reactions
**Short description:**
An AREA can have the possibility of having several Reactions.

**Describe the solution you'd like**
You need to modify the database to handle this case and make the multiple changes involved.

### LINKED ISSUE #131: Binary operator between Actions
**Short description:**
An ARea can be composed of multiple Actions. This actions are linked by binary operator (AND and OR).

**Describe the solution you'd like**
You need to modify the database to handle this case and make the multiple changes involved.
It can be only one operator per ARea.

### LINKED ISSUE #152: An AREA can have the possibility of having several Actions.
Short description:
An AREA can have the possibility of having several Actions.

Describe the solution you'd like
You need to modify the database to handle this case and make the multiple changes involved.

### PR DESCRIPTION
**Pull request by TomDUVAL-MAHE**
# Purpose

The purpose of this pull request is to update the database to have the possibility of having multiples actions and reaction for an AREA. It's also add the possibility to handle the logical gates 'OR' and 'AND' for an AREA

Closes #129
Closes #152
Closes #131

## Added features

* 2 AREA logical gates
  * OR : each action calls the reactions
  * AND : each action must be triggered once to call the reactions. After each reaction call, the status of each action is reset

## Updated features

* An AREA can now have multiple reactions
* An AREA can now have multiple actions
* New Database architechture
* The API has started to modify its architecture
  * New routes availlable
    * AREAs
      * **/api/area**
        * GET      
          * DO : Nothing
          * RETURN : List of every AREAs of the user making the request
        * POST
          * DO : Create an AREA
          * RETURN : The created AREA in JSON format
      * **/api/area/:id**
        * GET
          * DO : Nothing
          * RETURN : An AREA if the user have the right
        * PUT
          * DO : Update the AREA
          * RETURN : The updated AREA in JSON format
        * DELETE
          * DO : Delete the AREA
          * RETURN : The deleted AREA in JSON format
    * Reactions
      * **/api/area/:areaId/reactions**
        * GET
          * DO : Nothing
          * RETURN: List of every  Reaction/Parameter sets of the user making the request
        * POST
          * DO : Create a Reaction
          * RETURN: The created Reaction/Parameter set in JSON format
      * **/api/area/:areaId/reaction/:id**
        * GET
          * DO : Nothing
          * RETURN: A Reaction/Parameter set if the user have the right
        * PUT
          * DO : Update a Reaction
          * RETURN: Nothing 
        * DELETE
          * DO : Delete a Reaction
          * RETURN: The deleted Reaction/Parameter setin JSON format
    * Actions
      * **/api/area/:areaId/actions**
        * GET
          * DO : Nothing
          * RETURN: List of every  Action/Parameter sets of the user making the request
        * POST
          * DO : Create a Action
          * RETURN: The created Action/Parameter set in JSON format
      * **/api/area/:areaId/action/:id**
        * GET
          * DO : Nothing
          * RETURN: A Action/Parameter set if the user have the right
        * PUT
          * DO : Update a Action
          * RETURN: Nothing
        * DELETE
          * DO : Delete a Action
          * RETURN: The deleted Action/Parameter setin JSON format

### CODE REVIEW

File: application/app_code/lib/flutter_objects/reaction_data.dart [hunk 1]
```diff
@@ -43,36 +47,26 @@ class ReactionData {
...
-      }
-    }
-
-    return paramList;
+    return parametersContent;
   }
 
   /// Get a visual representation of a Reaction
   /// mode -> true = params, false = only text and desc
   /// params -> list of all the associated parameter content
```
Comment (tbellicha) [inline application/app_code/lib/flutter_objects/reaction_data.dart]: No more `params` parameter

**Review by tbellicha — CHANGES_REQUESTED** [review]
Just an old documentation

File: application/app_code/lib/flutter_objects/reaction_data.dart (hunk 1 above)
Comment (LucasTesnier) [reply application/app_code/lib/flutter_objects/reaction_data.dart]: fix

**Review by ZQUEMA — APPROVED** [review]

**Review by GuyomT — APPROVED** [review]
Too much code for a PR but NTR

**Review by LucasTesnier — APPROVED** [review]
Ouch

**Review by tbellicha — APPROVED** [review]

**Pull request merged**